### PR TITLE
fix(route53): warn CrossAccountZoneDelegationRecord delegatedZone Name Server undefined

### LINKED ISSUE #30600: aws-route53: `Not able to retrieve Name Servers for fake-name due to it being imported.` CrossAccountZoneDelegationRecord `delegatedZone` validation on NameServers blocks using Custom Resource to retrieve NameServers
Please add your +1 👍 to let us know you have encountered this
---

### Overview:
Versions 2.145.0-2.147.0 will throw a synthesis error when passing an imported `delegatedZone` to `route53.CrossAccountZoneDelegationRecord`. For example, the following code will throw:

```
   const parentZone = new route53.PublicHostedZone(stack, 'ParentHostedZone', {
      zoneName: 'myzone.com',
    });

    const childZone = route53.PublicHostedZone.fromPublicHostedZoneAttributes(stack, 'ChildHostedZone', {
      hostedZoneId: 'my-id',
      zoneName: 'my-name,
    });

 new route53.CrossAccountZoneDelegationRecord(stack, 'Delegation', {
      delegatedZone: childZone,
      parentHostedZoneId: parentZone.hostedZoneId,
      delegationRole: parentZone.crossAccountZoneDelegationRole,
    });
```

#### Complete Error Message:

```
Not able to retrieve Name Servers for my-name due to it being imported.
```

### Workaround:
Pin the version of `aws-cdk-lib` to `2.144.0`.

### Describe the bug

The fix in https://github.com/aws/aws-cdk/pull/30440 is blocking users from using custom resource to retrieve the delegated zone name servers.

### Expected Behavior

CDK should not throw error that blocks the user when the delegated zone is imported and does not have Name Servers information as users can use custom resource to retrieve the NS.

### Current Behavior

CrossAccountZoneDelegationRecord throws when `delegatedZone` is imported.

### Reproduction Steps

Following code will throw the error in question:
```
    const stack = new Stack();
    const parentZone = new route53.PublicHostedZone(stack, 'ParentHostedZone', {
      zoneName: 'myzone.com',
      crossAccountZoneDelegationPrincipal: new iam.AccountPrincipal('123456789012'),
    });

    // WHEN
    const childZone = route53.PublicHostedZone.fromPublicHostedZoneAttributes(stack, 'ChildHostedZone', {
      hostedZoneId: 'fake-id',
      zoneName: 'fake-name',
    });

    new route53.CrossAccountZoneDelegationRecord(stack, 'Delegation', {
      delegatedZone: childZone,
      parentHostedZoneId: parentZone.hostedZoneId,
      delegationRole: parentZone.crossAccountZoneDelegationRole,
    });
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.146.0

### Framework Version

_No response_

### Node.js Version

20

### OS

mac

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by samson-keung**
### Issue # (if applicable)

Closes #30600.

### Reason for this change

See https://github.com/aws/aws-cdk/pull/30440#issuecomment-2179987739

TLDR: This PR unblocks users who use other ways to retrieve delegated zone name servers

### Description of changes

Covert the `throw Error` into `addWarning`

### Description of how you validated changes

Unit test added

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by samson-keung** [comment]
Exemption Request

Reason: The change affects build time. No changes to deployment behaviour, hence, no integ test update

File: packages/aws-cdk-lib/aws-route53/lib/record-set.ts [hunk 1]
```diff
@@ -930,7 +930,8 @@ export class CrossAccountZoneDelegationRecord extends Construct {
     }
 
     if (!props.delegatedZone.hostedZoneNameServers) {
-      throw Error(`Not able to retrieve Name Servers for ${props.delegatedZone.zoneName} due to it being imported.`);
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-route53:importedDelegatedZoneNSValidation',
+        `Not able to retrieve Name Servers for \'${props.delegatedZone.zoneName}\' due to it being imported. Use a non-imported one or use AwsCustomResource to retrieve the Name Servers.`);
```
Comment (comcalvi) [inline packages/aws-cdk-lib/aws-route53/lib/record-set.ts:934]: how do we know this is imported? Not specified doesn't mean that it's been imported.

**Review by comcalvi — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-route53/lib/record-set.ts [hunk 2]
```diff
@@ -930,7 +930,8 @@ export class CrossAccountZoneDelegationRecord extends Construct {
     }
 
     if (!props.delegatedZone.hostedZoneNameServers) {
-      throw Error(`Not able to retrieve Name Servers for ${props.delegatedZone.zoneName} due to it being imported.`);
```
Comment (comcalvi) [inline packages/aws-cdk-lib/aws-route53/lib/record-set.ts:933]: Instead of throwing here we could change this line linked from the original issue: https://github.com/aws/aws-cdk/blob/d69c51a2e10864e69d4610b6df30d40a9bfec3ac/packages/aws-cdk-lib/aws-route53/lib/cross-account-zone-delegation-handler/index.ts#L58

to read:

```
 ResourceRecords: (DelegatedZoneNameServers ?? []).map(ns => ({ Value: ns })), 
```

but that change would need to be carefully integ tested, we've recently seen issues from passing empty arrays around the custom resource handler framework.

**Comment by TheRealAmazonKendra** [comment]
Quick FYI here, when you submit a fix, the title should describe the bug, not the fix.
